flexprep/domain/db_utils.py

```diff
@@ -54,53 +56,209 @@ def insert_item(self, item: IFSForecast) -> None:
+                combined_steps = self._prepare_processable_steps(step_zero_items, rows)
+
+                return combined_steps
+
+        except sqlite3.Error as e:
+            logger.exception(
+                f"An error occurred while querying the database for forecast_ref_time={forecast_ref_time}: {e}"
+            )
+            raise
+
+    def _fetch_step_zero_items(self, forecast_ref_time: dt) -> list:
         """
+        Fetch step-0 items from the database.
 
-        query = (
+        Args:
+            forecast_ref_time (datetime): The forecast reference time to query for.
+
+        Returns:
+            list: A list of step-0 rows.
+        """
+        step_zero_query = (
             "SELECT row_id, forecast_ref_time, step, key, processed "
             "FROM uploaded "
-            "WHERE forecast_ref_time = ?"
+            "WHERE forecast_ref_time = ? AND step = 0 "
+            "LIMIT 2"
         )
+        cursor = self.conn.execute(step_zero_query, (forecast_ref_time,))
+        return cursor.fetchall()
 
-        try:
-            with self.conn:
-                cursor = self.conn.execute(query, (forecast_ref_time,))
-                rows = cursor.fetchall()
-                logger.info(f"Query returned {len(rows)} items.")
-                results = [
+    def _fetch_current_and_previous_steps(self, forecast_ref_time: dt) -> list:
+        """
+        Fetch current steps and their previous steps from the database.
+
+        Args:
+            forecast_ref_time (datetime): The forecast reference time to query for.
+
+        Returns:
+            list: A list of rows containing both the current step and its previous step.
+        """
+        tstart = CONFIG.main.time_settings.tstart
+        tincr = CONFIG.main.time_settings.tincr
+
+        step_query = """
+        WITH step_data AS (
+            SELECT
+                cur.row_id AS cur_row_id,
+                cur.forecast_ref_time AS cur_forecast_ref_time,
+                cur.step AS cur_step,
+                cur.key AS cur_key,
+                cur.processed AS cur_processed,
+                prev.row_id AS prev_row_id,
+                prev.forecast_ref_time AS prev_forecast_ref_time,
+                prev.step AS prev_step,
+                prev.key AS prev_key,
+                prev.processed AS prev_processed,
+                ROW_NUMBER() OVER (PARTITION BY cur.step ORDER BY cur.step ASC) AS rn
+            FROM
+                uploaded cur
+            LEFT JOIN
+                uploaded prev
+            ON
+                cur.forecast_ref_time = prev.forecast_ref_time
+                AND prev.step = cur.step - ?
+            WHERE
+                cur.forecast_ref_time = ?
+                AND cur.processed = FALSE
+                AND cur.step != 0
+                AND (cur.step - ?) % ? = 0
+        )
+        SELECT *
+        FROM step_data
+        WHERE (prev_step = 0 AND rn = 1) OR (prev_step != 0) OR (prev_step is NULL)
```

Comment:
```suggestion
        WHERE (prev_step = 0 AND rn = 1) OR (prev_step != 0)
```
So that steps that do not have a previous step are skipped. The downside being that the debug message is lost but this allows to simplify the logic in the loop further (not having the middle continue statement)